p2p/exchange.go

```diff
@@ -124,16 +124,25 @@ func (ex *Exchange[H]) Head(ctx context.Context) (H, error) {
 		defer cancel()
 	}
 
+	reqParams := DefaultRequestParams()
+	for _, opt := range reqOpts {
+		opt(&reqParams)
+	}
+
+	peers := ex.peerTracker.GetPeers()
```

Comment:
Maybe we can reduce the number of peers here? If peerTracker will be full, then we will send 100 requests.

Reply:
Oh good catch, I meant to take 5 or so

Reply:
https://github.com/celestiaorg/go-header/pull/53/commits/b3b9299fe48ee046761d3f45f8408fa2b825331b